src/Tree/index.tsx

```diff
@@ -377,6 +383,43 @@ class Tree extends React.Component<TreeProps, TreeState> {
     }
   };
 
+  /**
+   * Takes a hierarchy point node and centers the node on the screen
+   * if the dimensions parameter is passed to the tree.
+   *
+   * This code is adapted from Rob Schmuecker's centerNode method.
+   * Link: http://www.robschmuecker.com/d3-js-drag-and-drop-zoomable-tree/
+   *
+   * @param hierarchyPointNode
+   */
+  centerNode = (hierarchyPointNode: HierarchyPointNode<TreeNodeDatum>) => {
+    const g = select(`.${this.gInstanceRef}`);
+    const svg = select(`.${this.svgInstanceRef}`);
+    const scale = this.state.d3.scale;
+    //if the dimensions are given
+    if (this.props.dimensions) {
```

Comment:
Can we move this check to the top of the method so it's wrapping everything? Otherwise the `select` query and `const` assignments will happen unnecessarily on every node click if `dimensions` is not defined.

Reply:
True, will move up